packages/react-tinacms-strapi/src/strapi-provider.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react'
+
+import React from 'react'
+import { StrapiAuthenticationModal } from './strapi-auth-modal'
+import { useCMS } from 'tinacms'
+
+interface ProviderProps {
+  children: any
+  onLogin: () => void
+  onLogout: () => void
+}
+export const StrapiProvider = ({
+  children,
+  onLogin,
+  onLogout,
+}: ProviderProps) => {
+  const [activeModal, setActiveModal] = useState('none')
+
+  const beginAuth = async () => {
+    setActiveModal('authenticate')
+  }
+
+  const onClose = async () => {
+    setActiveModal('none')
+  }
+
+  const onAuthSuccess = async () => {
+    onLogin()
+    setActiveModal('none')
+  }
```

Comment:
These don't need to be async

```suggestion
  const beginAuth = () => {
    setActiveModal('authenticate')
  }

  const onClose = () => {
    setActiveModal('none')
  }

  const onAuthSuccess = () => {
    onLogin()
    setActiveModal('none')
  }
```